crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -77,6 +77,7 @@ impl<F: Send + Clone + Sync> GetTracker<F, PaymentData<F>, api::PaymentsRequest>
         key_store: &domain::MerchantKeyStore,
         auth_flow: services::AuthFlow,
         header_payload: &hyperswitch_domain_models::payments::HeaderPayload,
+        _platform_merchant_account: Option<&domain::MerchantAccount>,
```

Comment:
we should validate that this belongs to the same merchant_id that was passed in create right?

Reply:
yess, will add validations in eventual PRs.